lib/Controller/LoginController.php

```diff
@@ -305,6 +303,37 @@ public function login(int $providerId, string $redirectUrl = null) {
 		return new RedirectResponse($authorizationUrl);
 	}
 
+	/**
+	 * @param string $authorizationEndpoint
+	 * @param array $extraGetParameters
+	 * @return string
+	 */
+	private function buildAuthorizationUrl(string $authorizationEndpoint, array $extraGetParameters = []): string {
+		$parsedUrl = parse_url($authorizationEndpoint);
+
+		$urlWithoutParams =
+			(isset($parsedUrl['scheme']) ? $parsedUrl['scheme'] . '://' : '')
+			. ($parsedUrl['host'] ?? '')
+			. (isset($parsedUrl['port']) ? ':' . $parsedUrl['port'] : '')
+			. ($parsedUrl['path'] ?? '');
+
+		$queryParams = $extraGetParameters;
+		if (isset($parsedUrl['query'])) {
+			$params = explode('&', $parsedUrl['query']);
+			foreach ($params as $p) {
+				$keyValue = explode('=', $p);
+				if (count($keyValue) === 2) {
+					$queryParams[$keyValue[0]] = $keyValue[1];
```

Comment:
Should we still call htmlentities on the key and value?

Reply:
https://github.com/nextcloud/user_oidc/pull/518/files#diff-35a02217ab68218fc7a45f4a7b5483d63a0ea6b80973c2938f73f181845178a0R333

As the final URL is encoded, probably not needed?

I am still unsure whether it would not be better to move the `htmlentities()` to the following line where the HTML is crafted, though. 🙈

Reply:
@juliushaertl Like Php doc says:

> http_build_query — Generate URL-encoded query string

So it seems enough to me.

@lauritzh 

> I am still unsure whether it would not be better to move the htmlentities() to the following line where the HTML is crafted, though.

We can't because, for example, it will replace `&` by `&amp;` in the query string.

Reply:
Ahh I see. Thank you very much for clarification :)